### PR TITLE
Update user data  and delete user

### PR DESCRIPTION
**Pull request by Bino26**
Hello tean, 
I create today endpoints `/updateuser` and `/deleteuser` on demand from the frontend team .
The next thing to do with user will be to work on password functionnality(reset,forgot password...) and delete the user will all his data(notes,answers,....)

### CODE REVIEW

File: src/controllers/userController.js [hunk 1]
```diff
@@ -100,4 +100,54 @@ const logOut = (req, res) => {
   }
 };
 
-module.exports = { signUp, logIn, logOut };
+/******************************************************
+ * @UPDATEUSER
+ * @route /api/v1/updateuser
```
Comment (in43sh) [inline src/controllers/userController.js:161]: I'm really curious what is type of comment called? And where are you getting all these keywords (`@route`, `@method`, `@description`, `@body`, `@returns`) from?

File: src/controllers/userController.js [hunk 2]
```diff
@@ -100,4 +100,54 @@ const logOut = (req, res) => {
...
+ * @method PUT
+ * @description singUp function for creating new user
+ * @body name, email, password, confirmPassword
```
Comment (in43sh) [inline src/controllers/userController.js]: If you put `name`, `email`, `password`, `confirmPassword` in `@body`, does it mean that you have pass them in `req.body`? In that case, do you need to pass `req.user`?

File: src/routes/mainRouter.js [hunk 3]
```diff
@@ -9,5 +15,7 @@ const router = express.Router();
 router.post("/signup", signupDataValidate, signUp);
 router.route("/login").post(loginDataValidate, logIn).get(authenticateUser);
 router.get("/logout", logOut);
+router.put("/updateuser", authenticateUser, updateUser);
```
Comment (in43sh) [inline src/routes/mainRouter.js:23]: I would like to recommend to use a route naming convention that was suggest by Elijah in Slack. Let me know if you have questions about it

File: src/controllers/userController.js (hunk 1 above)
Comment (in43sh) [reply src/controllers/userController.js:161]: I searched and found that there is [JSDoc](https://jsdoc.app/). But I couldn't find `@route`, `@method`, `@description`, `@body`, `@returns` in the allowed methods. Could you clarify it?

File: src/controllers/userController.js [hunk 4]
```diff
@@ -100,4 +100,54 @@ const logOut = (req, res) => {
...
+ * @returns User Object
+ ******************************************************/
+
+const updateUser = async (req, res) => {
+  try {
+    const { userId } = req.user; //retrieve user.id from jwtverify()
```
Comment (in43sh) [inline src/controllers/userController.js:170]: Could you clarify where `jwtverify()` comes from?

Comment (in43sh) [reply src/controllers/userController.js:170]: Also, it's not a part of `req.body`. Could you explain how you get access to `req.user`?

File: src/controllers/userController.js [hunk 5]
```diff
@@ -100,4 +100,54 @@ const logOut = (req, res) => {
...
+    const user = await userModel.findByIdAndUpdate(userId, req.body);
```
Comment (in43sh) [inline src/controllers/userController.js]: If we pass user password in body and just update it in the database then it gets updated and becomes the new password, when it should get hashed before being sent to the database.

File: src/controllers/userController.js (hunk 2 above)
Comment (in43sh) [inline src/controllers/userController.js]: Do we need to send `confirmPassword` if we are not doing any password confirmation? Also, `User` model doesn't have `confirmPassword` field, so we can remove it.

**Review by in43sh — CHANGES_REQUESTED** [review]
Hi Bino! Overall, nice job creating controllers controllers `updateUser` and `deleteUser`. They are working but there are some changes that I wanted you to look at. Please, let me know if you have any questions.
Also, could you provide some examples of what is the right way to make requests?

**Comment by in43sh** [comment]
Also, I noticed some strange behavior when trying to sign up. Let's say we just created a new user with the following credentials:
```
{
    "firstname": "test",
    "lastname": "test",
    "username": "test",
    "email": "test@test.com",
    "password": "test"
}
```
We used this route `http://localhost:8000/api/v1/signup`
Now, let's create a new user with only `username` and `email` being different:
```
{
    "firstname": "test",
    "lastname": "test",
    "username": "test1",
    "email": "test@test.com1",
    "password": "test"
}
```
It will give an error:
```
{
    "success": false,
    "message": "User already exists"
}
```
In my opinion, it's strange because in the `User` model only 2 fields are marked as unique, so it shouldn't happen. @Bino26, please correct me if I'm wrong.

**Comment by in43sh** [comment]
Also, let's say I just did a request to `http://localhost:8000/api/v1/deleteuser` and send the response back as:
```
{
    "success": true,
    "message": "User deleted successfuly"
}
```
But then, if we try doing it again, it will return the same response when it should say that the user was already deleted.

File: src/controllers/userController.js (hunk 1 above)
Comment (Bino26) [reply src/controllers/userController.js:161]: It's not method😅 😅
 Just a way to comment my code .

**Comment by Bino26** [comment]
About differents username and email, it's about dupplicate error 11000.
MongoDb doesn't accept this .
E11000 duplicate key error collection: Quizzes.users index: firstname_1 dup key: { firstname: \"Yann\" }"
You can do your research about this
This is the error when you try the samething . 
Maybe i will change the message has "first name , lastname or email has already exist

**Comment by Bino26** [comment]
About the delete , the bug was about cookies .
They were not deleted when user was deleted .
I have fixed now .

**Review by in43sh — APPROVED** [review]

**Pull request merged**